File: src/main/java/source/KafkaSource.java
package source;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer010;

import java.util.Properties;

public class KafkaSource {

    public static void main(String[] args) throws Exception {

        //流编程环境
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        //kafka配置
        Properties props =new Properties();
        props.put("bootstrap.servers", "localhost:9092");
        props.put("zookeeper.connect", "localhost:2181");
        props.put("group.id", "metric-group");
        props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");  //key 反序列化
        props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        props.put("auto.offset.reset", "latest"); //value 反序列化

        //添加source
        DataStream<String> dataStream = env.addSource(new FlinkKafkaConsumer010<String>("",new SimpleStringSchema(),props)).setParallelism(1);

        //打印数据
        dataStream.print();

        env.execute("Flink kafka source");

    }
}

File: src/main/scala/com/taolu/flink/source/KafkaSource.scala
package com.taolu.flink.source

import java.util.Properties

object KafkaSource {

  //创建kafka配置
  def getProperties():Properties = {
    val pro = new Properties()
    pro.setProperty("bootstrap.servers","127.0.0.1:9092")
    pro.setProperty("group.id","lutaoConsumer")
    pro.setProperty("key.deserializer","org.apache.kafka.common.serialization.StringDeserializer")
    pro.setProperty("value.deserializer","org.apache.kafka.common.serialization.StringDeserializer")
    pro.setProperty("enable.auto.commit","false")
    pro.setProperty("auto.offset.reset","earliest")
    pro
  }

}
